packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -3085,29 +3085,14 @@ export class StripeHelper extends StripeHelperBase {
       number: invoiceNumber,
       currency: paymentProratedCurrency,
       total: invoiceTotalNewInCents,
+      amount_due: invoiceAmountDue,
     } = invoice;
 
-    // Using stripes default proration behaviour
-    // (https://stripe.com/docs/billing/subscriptions/upgrade-downgrade#immediate-payment)
-    // an invoice is only created at time of upgrade, if the billing period changes.
-    // In the case that the billing period is the same, we sum the pending invoiceItems
-    // retrieved in upcomingInvoiceWithInvoiceitem.
-    // The actual recuring charge, for the next billing cycle, shows up as a type = 'subscription'
-    // line item.
-    const onetimePaymentAmount = upcomingInvoiceWithInvoiceitem
```

Comment:
Is this logic no longer necessary? Looking at the PR that implemented, this logic was to account for upgrades where the interval does not change  (e.g. Month to month) and there are pending invoice items on the customer.

Reply:
I am open to changing this. Regarding same billing cycle prorations, Stripe adds the list price and proration (without tax) and then applies the tax on top. The problem with this is how we word the email. It says that the `original list price + tax` will upgrade to the `new list price + tax`. It then says "you will also be charged," which is an additional amount on top of the new total.

The previous code was implemented for same billing cycle but omitted tax. This made the upcoming invoice amounts not add up correctly. We could change the wording to say "not including tax" or "plus tax" for the proration portion, but I opted to try and correct this, which made the above code no longer necessary.

I would prefer not to do this and leave the calculations to Stripe, but could not see another way. I couldn't find the actual tax percentage being applied anywhere. Only amount after tax is calculated, so I couldn't apply tax to the above code and instead chose to get the difference instead, as seen in the revised code.

Note: I saw the other comment and will double check the math because I did forget to test for inclusive vs exclusive behaviour, and testing what shows up with how it's handled in other countries.